ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/SupportsSanityCheck.hs

```diff
@@ -0,0 +1,83 @@
+-- | This module adds support for sanity checking consensus configuration
+--   on node startup. These checks should primarily look for unusual
+--   configuration choices that may point to an accidentally-misconfigured node
+--   and quietly cause problems, rather than incoherent configurations that will
+--   result in fatal errors at a later point.
+--
+--   While in most situations they can be handled as fatal issues, there are
+--   situations when intentionally configuring a node "weirdly" can be useful,
+--   and so the user should be able to opt out of the sanity checks at their
+--   own peril.
+module Ouroboros.Consensus.Block.SupportsSanityCheck (
+    BlockSupportsSanityCheck (..)
+  , SanityCheckIssue (..)
+  , checkSecurityParamConsistency
+  , issueDescription
+  , sanityCheckConfig
+  ) where
+
+import           Control.Exception
+import           Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NonEmpty
+import           Data.Maybe (catMaybes)
+import           Ouroboros.Consensus.Config (TopLevelConfig)
+import           Ouroboros.Consensus.Config.SecurityParam
+
+-- | An issue found in the 'TopLevelConfig' for a block. See 'issueDescription'
+--   for human-readable descriptions of each of these cases, especially when
+--   presenting these to users.
+data SanityCheckIssue
+  -- | Configuration contains multiple security parameters. This may cause
+  --   strange behaviour around era boundaries.
+  = InconsistentSecurityParam (NonEmpty SecurityParam)
+  deriving (Show, Eq)
+
+instance Exception SanityCheckIssue
+
+issueDescription :: SanityCheckIssue -> String
+issueDescription (InconsistentSecurityParam ks) =
+  mconcat
+    [ "SecurityParams (K) were found to be inconsistent between constituent "
+    , "eras of a HardForkBlock: "
+    , show (NonEmpty.toList ks)
+    ]
```

Comment:
Maybe this could be the implementation of `displayException` in addition to/instead of a separate function?